internal/operators/odf/odf_operator.go

```diff
@@ -218,47 +241,30 @@ func (o *operator) GetHostRequirements(_ context.Context, cluster *common.Cluste
 
 		/* getValidDiskCount counts the total number of valid disks in each host and return a error if we don't have the disk of required size,
 		we ignore the error as its treated as 500 in the UI */
-		diskCount, _ = o.getValidDiskCount(inventory.Disks, host.InstallationDiskID, &cluster.Cluster, nil)
+		diskCount, _ = o.getValidDiskCount(inventory.Disks, host.InstallationDiskID, nil, mode)
 	}
 
-	role := common.GetEffectiveRole(host)
-
+	// master / auto-assign in compact mode
 	if mode == compactMode {
 		var reqDisks int64 = 1
 		if diskCount > 0 {
 			reqDisks = diskCount
 		}
-		// for each disk odf requires 2 CPUs and 5 GiB RAM
-		if role == models.HostRoleMaster || role == models.HostRoleAutoAssign {
-			return &models.ClusterHostRequirementsDetails{
-				CPUCores: o.config.ODFPerHostCPUCompactMode + (reqDisks * o.config.ODFPerDiskCPUCount),
-				RAMMib:   conversions.GibToMib(o.config.ODFPerHostMemoryGiBCompactMode + (reqDisks * o.config.ODFPerDiskRAMGiB)),
-			}, nil
-		}
-		// regular worker req
-		return &models.ClusterHostRequirementsDetails{
-			CPUCores: o.config.ODFPerHostCPUStandardMode + (reqDisks * o.config.ODFPerDiskCPUCount),
-			RAMMib:   conversions.GibToMib(o.config.ODFPerHostMemoryGiBStandardMode + (reqDisks * o.config.ODFPerDiskRAMGiB)),
-		}, nil
-	}
-
-	// In standard mode, ODF does not run on master nodes so return zero
-	if role == models.HostRoleMaster {
-		return &models.ClusterHostRequirementsDetails{CPUCores: 0, RAMMib: 0}, nil
-	}
 
-	// worker and auto-assign
-	if diskCount > 0 {
-		// for each disk odf requires 2 CPUs and 5 GiB RAM
+		// Each ODF disk requires 2 CPUs and 5 GiB RAM
 		return &models.ClusterHostRequirementsDetails{
-			CPUCores: o.config.ODFPerHostCPUStandardMode + (diskCount * o.config.ODFPerDiskCPUCount),
-			RAMMib:   conversions.GibToMib(o.config.ODFPerHostMemoryGiBStandardMode + (diskCount * o.config.ODFPerDiskRAMGiB)),
+			CPUCores: o.config.ODFPerHostCPUCompactMode + (reqDisks * o.config.ODFPerDiskCPUCount),
```

Comment:
What's the difference between host in compact mode and worker in standard mode?
Why here we set it as `reqDisks` and below as `diskCount`? Is it because we can not run ODF if a node has no disk?

In this case, wouldn't something like this be clearer?

```suggestion
// Workers can have 0 requirements if no disk because reasons
d := diskCount
cpuPerHost = o.config.ODFPerHostCPUCompactMode
perHostRAMGiB = o.config.ODFPerHostMemoryGiBCompactMode

if mode == compactMode {
   d = math.Max(reqDisks, diskCount)
   cpuPerHost = o.config.ODFPerHostCPUStandardMode
   perHostRAMGiB = o.config.ODFPerHostMemoryGiBStandardMode
}

return &models.ClusterHostRequirementsDetails{
		CPUCores: cpuPerHost + (d * o.config.ODFPerDiskCPUCount),
		RAMMib:   conversions.GibToMib(o.config.perHostRAMGiB + (d * o.config.ODFPerDiskRAMGiB)),
	}, nil
```

Reply:
This is an excellent question. This was the behaviour before my changes, and as I couldn't get to the bottom of this and This is not really related to my PR, I preferred leaving this as is.

Reply:
I understand, however we are already simplifying the code that was there before by removing unnecessary if in the standard mode, we probably could go a step further and simplify it even more, WDYT?